src/Mod/Assembly/CommandCreateSimulation.py

```diff
@@ -0,0 +1,1019 @@
+        label = label + ")"
+        self.app_obj.Label = label
+
+
+class MotionEditDialog:
+    def __init__(self, assembly, motionType=MotionTypes[0], joint=None, formula="5*time"):
+        self.assembly = assembly
+        self.motionType = motionType
+        self.joint = joint
+        self.formula = formula
+
+        # Create a non-modal, frameless dialog
+        self.dialog = QDialog()
+        self.dialog.setWindowFlags(Qt.Popup)
+        self.initialPos = QCursor.pos()
+        self.dialog.setMinimumSize(500, 200)  # Set a reasonable minimum size
+
+        # Create the joints combobox
+        self.joint_combo = QComboBox(self.dialog)
+        self.setup_joint_combo()
+
+        # Create the motion type combobox
+        self.motion_type_combo = QComboBox(self.dialog)
+        self.setup_motiontype_combo()
+
+        def on_motion_type_changed(text):
+            self.motionType = text
+
+        self.motion_type_combo.currentTextChanged.connect(on_motion_type_changed)
+
+        def on_joint_changed(index):
+            self.joint = self.joint_combo.itemData(index)
+            self.setup_motiontype_combo()  # Refresh the motion combo box based on the new joint type
+
+        self.joint_combo.currentIndexChanged.connect(on_joint_changed)
+
+        # Create the line edit for the formula
+        formula_edit = QLineEdit(self.dialog)
+        formula_edit.setText(self.formula)
+        formula_edit.setPlaceholderText(translate("Assembly", "Enter your formula..."))
+
+        # Connect the line edit to update the Formula property
+        def on_formula_changed(text):
+            self.formula = text
+
+        formula_edit.textChanged.connect(on_formula_changed)
+
+        self.setupHelpSection()
+
+        # Create Ok and Cancel buttons
+        button_box = QDialogButtonBox(
+            QDialogButtonBox.Ok | QDialogButtonBox.Cancel, Qt.Horizontal, self.dialog
+        )
+        button_box.accepted.connect(self.dialog.accept)
+        button_box.rejected.connect(self.dialog.reject)
+
+        # Set up the layout of the dialog
+        layout = QGridLayout(self.dialog)
+
+        # Add labels and widgets to the layout
+        layout.addWidget(QLabel("Joint:"), 0, 0)
+        layout.addWidget(self.joint_combo, 0, 1)
+
+        layout.addWidget(QLabel("Motion Type:"), 1, 0)
+        layout.addWidget(self.motion_type_combo, 1, 1)
+
+        layout.addWidget(QLabel("Formula:"), 2, 0)
+        layout.addWidget(formula_edit, 2, 1)
+
+        # Add the help label above the buttons
+        layout.addWidget(self.help_label0, 3, 0, 1, 2)
+        layout.addWidget(self.help_label1, 4, 0, 1, 2)
+        layout.addWidget(self.help_label2, 5, 0, 1, 2)
+        layout.addWidget(self.help_label3, 6, 0, 1, 2)
+        layout.addWidget(self.help_label4, 7, 0, 1, 2)
+        layout.addWidget(self.help_label5, 8, 0, 1, 2)
+        layout.addWidget(self.help_label6, 9, 0, 1, 2)
+        layout.addWidget(self.help_label7, 10, 0, 1, 2)
```

Comment:
Can we give better names to those fields?

Reply:
I don't know, I feel there is more useful information in the numbers than there would be in the names

Reply:
Fine, looking at code below I think that every naming scheme would have pros and cons :sweat_smile:. Feel free to resolve, I'm not sure how to do that on GH as reviewer...